projects/RabbitMQ.Client/client/impl/ChannelBase.cs

```diff
@@ -450,16 +450,22 @@ protected ValueTask ModelSendAsync<T>(in T method, CancellationToken cancellatio
         }
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        protected ValueTask ModelSendObserveFlowControlAsync<TMethod, THeader>(in TMethod method, in THeader header, ReadOnlyMemory<byte> body, CancellationToken cancellationToken)
+        protected ValueTask ModelSendAsync<TMethod, THeader>(in TMethod method, in THeader header, ReadOnlyMemory<byte> body, CancellationToken cancellationToken)
```

Comment:
async methods cannot have in parameters. Because I wanted to preserve that I have decided to move the flow control logic one level up. Not perfect but it felt better than removing the in modifiers.

Reply:
I would simply get rid of the in modifier as we did in #1601.

Reply:
The PR you pointed out refers to the public API surface while this is concerning the inner implementation details of the protocol interaction. These don't necessarily have to be treated the same